pkg/kernel/kernel.go

```diff
@@ -0,0 +1,52 @@
+package kernel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alibaba/pouch/pkg/exec"
+	"github.com/pkg/errors"
+)
+
+// VersionInfo holds information about the kernel.
+type VersionInfo struct {
+	Kernel int    // Version of the kernel (e.g. 4.1.2-generic -> 4)
+	Major  int    // Major part of the kernel version (e.g. 4.1.2-generic -> 1)
+	Minor  int    // Minor part of the kernel version (e.g. 4.1.2-generic -> 2)
+	Flavor string // Flavor of the kernel version (e.g. 4.1.2-generic -> generic)
+}
+
+// String returns the kernel version's string format.
+func (k *VersionInfo) String() string {
+	return fmt.Sprintf("%d.%d.%d-%s", k.Kernel, k.Major, k.Minor, k.Flavor)
+}
+
+// GetKernelVersion returns the kernel version info.
+func GetKernelVersion() (*VersionInfo, error) {
+	var (
+		kernel, major, minor int
+		flavor               string
+	)
+
+	_, stdout, _, err := exec.Run(0, "uname", "-a")
```

Comment:
maybe you could use `uname -r`?

Reply:
you are right!! Thank you for your reminder.